integration-tests/proxy_test.go

```diff
@@ -774,3 +775,166 @@ func TestProxyNexctlConnections(t *testing.T) {
+		require.NoError(err)
+		_, err := helper.containerExec(ctx, serverNode, []string{"python3", "-c", "import os; open('/tmp/bananas/index.html', 'w').write('bananas')"})
+		require.NoError(err)
+		_, _ = helper.containerExec(ctx, serverNode, []string{"python3", "-m", "http.server", "-b", "::", "-d", "/tmp/bananas", "8081"})
+	})
+	util.GoWithWaitGroup(&wg, func() {
+		output, _ := helper.containerExec(ctx, serverNode, []string{"udpong", "4240", "carrot"})
+		helper.Logf("udpong output: %s", output)
+	})
+	util.GoWithWaitGroup(&wg, func() {
+		_, _ = helper.containerExec(ctx, serverNode, []string{"udpong", "4241", "potato"})
+	})
+
+	// run clients on clientNode to reach the server on serverNode
+	dialIP := opts.dialIP(node2IP)
+
+	ctxTimeout, curlCancel := context.WithTimeout(ctx, 10*time.Second)
+	defer curlCancel()
+	success, err := util.CheckPeriodically(ctxTimeout, time.Second, func() (bool, error) {
+		bananas := 0
+		apples := 0
+		carrot := 0
+		potato := 0
+		for i := 0; i < 10; i++ {
+
+			output, err := helper.containerExec(ctx, clientNode, []string{"curl", "-s", fmt.Sprintf("http://%s:80", dialIP)})
+			if err != nil {
+				helper.Logf("Retrying curl for up to 10 seconds: %v -- %s", err, output)
+				return false, nil
+			}
+			if strings.Contains(output, "bananas") {
+				bananas += 1
+			} else if strings.Contains(output, "apples") {
+				apples += 1
+			}
+
+			output, err = helper.containerExec(ctx, clientNode, []string{"udping", dialIP, "42"})
+			if err != nil {
+				helper.Logf("Retrying udp client for up to 10 seconds: %v -- %s", err, output)
+				return false, nil
+			}
+			if strings.Contains(output, "carrot") {
+				carrot += 1
+			} else if strings.Contains(output, "potato") {
+				potato += 1
+			}
+		}
+		require.Equal(5, apples)
+		require.Equal(5, bananas)
+		require.Equal(5, carrot)
+		require.Equal(5, potato)
+		return true, nil
+	})
+	require.NoError(err)
+	require.True(success)
+	_, _ = helper.containerExec(ctx, serverNode, []string{"killall", "python3"})
+	_, _ = helper.containerExec(ctx, serverNode, []string{"killall", "udpong"})
+	wg.Wait()
+}
+
+func TestProxyLoadBalancer(t *testing.T) {
+	tests := []testProxyLoadBalancerOpts{
+		{
+			name: "Egress",
+			flag: "--egress",
+			upstreamIP: func(node1IP string) string {
+				return node1IP
+			},
+			dialIP: func(node2IP string) string {
+				return "127.0.0.1"
+			},
+			workloadPlacer: func(node1, node2 testcontainers.Container) (serverNode testcontainers.Container, clientNode testcontainers.Container) {
+				return node1, node2
+			},
+		},
+		{
+			name: "Ingress",
+			flag: "--ingress",
+			upstreamIP: func(node1IP string) string {
+				return "127.0.0.1"
+			},
+			dialIP: func(node2IP string) string {
+				return node2IP
+			},
+			workloadPlacer: func(node1, node2 testcontainers.Container) (serverNode testcontainers.Container, clientNode testcontainers.Container) {
+				return node2, node1
+			},
+		},
+	}
```

Comment:
How about a couple more test cases that use IPv6 addresses? I spotted a couple of IPv6 address-handling bugs in the code, so I think it's worth adding the coverage here. I'm not sure if these tests would hit what I saw, but it's still good to add, I think.

Reply:
will do.